crates/nu-lsp/src/lib.rs

```diff
@@ -280,11 +294,20 @@ impl LanguageServer {
                     if let Some(span) = &block.span {
                         for cached_file in working_set.files() {
                             if cached_file.covered_span.contains(span.start) {
+                                let uri = Url::from_file_path(&*cached_file.name).ok()?;
+                                let rope_of_file = self
+                                    .ropes
+                                    .entry(uri.to_file_path().ok()?)
+                                    .or_insert_with(|| {
```

Comment:
Do you have an idea how that then plays together with the [`textDocument` synchronization](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_synchronization)?

> The document’s content is now managed by the client and the server must not try to read the document’s content using the document’s Uri.

Maybe, the code shouldn't be duplicated in the match arms.

Reply:
I think DidChange is handled properly in notification.rs since you have to first open it to change it, by opening it, a refresh on its content is not redundant since it might get changed elsewhere, while DidOpen is tricky since some cached files brought from engine state by `nu -I` will not trigger it, therefore, have to deal with empty entries here.